src/lib/infrastructure/server/trpc/routers/research-contexts/list-research-contexts-router.ts

```diff
@@ -0,0 +1,33 @@
+import { Logger } from "pino";
+import serverContainer from "../../../config/ioc/server-container";
+import { createTRPCRouter, protectedProcedure } from "../../server";
+import { CONTROLLERS, UTILS } from "../../../config/ioc/server-ioc-symbols";
+import { TListResearchContextsViewModel } from "~/lib/core/view-models/list-research-contexts-view-models";
+import { Signal } from "~/lib/core/entity/signals";
+import ListResearchContextsController from "../../../controller/list-research-contexts-controller";
+import { SIGNAL_FACTORY } from "~/lib/infrastructure/common/signals-ioc-container";
+
+const getLogger = () => {
+    const loggerFactory = serverContainer.get<(module: string) => Logger>(UTILS.LOGGER_FACTORY);
+    const logger = loggerFactory("conversationRouter");
+    return logger;
+}
+
+export const listResearchContextsRouter = createTRPCRouter({
+    listAll: protectedProcedure
+      .query(async () => {
+        const loggerFactory = serverContainer.get<(module: string) => Logger>(UTILS.LOGGER_FACTORY);
+        const logger = loggerFactory("List Research Contexts TRPC Router");
+
+        const signalFactory = serverContainer.get<(initialValue: TListResearchContextsViewModel, update?: (value: TListResearchContextsViewModel) => void) => Signal<TListResearchContextsViewModel>>(SIGNAL_FACTORY.KERNEL_LIST_RESEARCH_CONTEXTS);
```

Comment:
Here a signalsContainer should be used instead, as otherwise the symbol cannot be found